orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/pipeline/RunCanaryIntervalsStage.kt

```diff
@@ -46,6 +47,18 @@ class RunCanaryIntervalsStage(private val clock: Clock) : StageDefinitionBuilder
   override fun taskGraph(stage: Stage, builder: TaskNode.Builder) {
   }
 
+  private fun getDeployDetails(stage: Stage) : DeployedServerGroupContext? {
+    val deployedServerGroupsStage = stage.parent?.execution?.stages?.find {
```

Comment:
IMO, you'll want to use `StageNavigator` and not just look for the first matching stage in the execution.

The navigator will look up the graph and allow you to grab the "closest" stage.

Reply:
I'll look into this, thanks

Reply:
Scratch that comment ... misunderstanding around the shape of this graph.